airflow/contrib/kubernetes/worker_configuration.py

```diff
@@ -189,9 +189,9 @@ def make_pod(self, namespace, worker_uuid, pod_id, dag_id, task_id, execution_da
             limit_cpu=kube_executor_config.limit_cpu
         )
         gcp_sa_key = kube_executor_config.gcp_service_account_key
-        annotations = {
-            'iam.cloud.google.com/service-account': gcp_sa_key
-        } if gcp_sa_key else {}
+        annotations = kube_executor_config.annotations
```

Comment:
I'm not sure if this is a problem or not - but this will (when gcp_sa_key is set) alter the hash in place.

Does that matter? If so `kube_executor_config.annotations.copy()` is the fix.